Lib/pathlib/types.py

```diff
@@ -14,10 +14,15 @@
 from glob import _PathGlobber
 from pathlib._os import magic_open, ensure_distinct_paths, ensure_different_files, copyfileobj
 from pathlib import PurePath, Path
-from typing import Optional, Protocol, runtime_checkable
+from typing import (
+    Any, BinaryIO, Callable, Generator, Iterator, Literal, Optional, Protocol, Self, Sequence, TypeVar,
```

Comment:
For 3.9+, we can use `from collections.abc import Callable, Iterator, ...` etc.

Reply:
I moved for all deprecated typing ABCs to collections.abc